turbo/jsonrpc/bor_snapshot.go

```diff
@@ -193,6 +199,96 @@ func (api *BorImpl) GetCurrentValidators() ([]*valset.Validator, error) {
 	return snap.ValidatorSet.Validators, nil
 }
 
+func (api *BorImpl) SendRawTransactionConditional(ctx context.Context, encodedTx hexutility.Bytes, options types.OptionsAA4337) (common.Hash, error) {
+	txn, err := types.DecodeWrappedTransaction(encodedTx)
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	txTemp, err := api.db.BeginRo(ctx)
+	if err != nil {
+		return common.Hash{}, err
+	}
+	defer txTemp.Rollback()
+
+	currentHeader := rawdb.ReadCurrentHeader(txTemp)
+
+	tempBlockNumber := rpc.BlockNumber(currentHeader.Number.Int64())
+	tempBlockorHash := rpc.BlockNumberOrHash{BlockNumber: &tempBlockNumber}
+
+	readerTemp, err := rpchelper.CreateStateReader(ctx, txTemp, tempBlockorHash, 0, api.filters, api.stateCache, api.historyV3(txTemp), "")
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	currentState := state.New(readerTemp)
+
+	// check block number range
+	if err := currentHeader.ValidateBlockNumberOptions4337(options.BlockNumberMin, options.BlockNumberMax); err != nil {
+		return common.Hash{}, &rpc.OptionsValidateError{Message: "out of block range. err: " + err.Error()}
+	}
+
+	// check timestamp range
+	if err := currentHeader.ValidateTimestampOptions4337(options.TimestampMin, options.TimestampMax); err != nil {
+		return common.Hash{}, &rpc.OptionsValidateError{Message: "out of time range. err: " + err.Error()}
+	}
+
+	// check knownAccounts length (number of slots/accounts) should be less than 1000
+	if err := options.KnownAccounts.ValidateLength(); err != nil {
+		return common.Hash{}, &rpc.KnownAccountsLimitExceededError{Message: "limit exceeded. err: " + err.Error()}
+	}
+
+	// check knownAccounts
+	if err := currentState.ValidateKnownAccounts(options.KnownAccounts); err != nil {
+		return common.Hash{}, &rpc.OptionsValidateError{Message: "storage error. err: " + err.Error()}
+	}
+
+	// put options data in Tx, to use it later while block building
+	txn.PutOptions(&options)
+
+	// If the transaction fee cap is already specified, ensure the
+	// fee of the given transaction is _reasonable_.
+	if err := checkTxFee(txn.GetPrice().ToBig(), txn.GetGas(), ethconfig.Defaults.RPCTxFeeCap); err != nil {
+		return common.Hash{}, err
+	}
+	if !txn.Protected() {
+		return common.Hash{}, errors.New("only replay-protected (EIP-155) transactions allowed over RPC")
+	}
+
+	// this has been moved to prior to adding of transactions to capture the
+	// pre state of the db - which is used for logging in the messages below
+	tx, err := api.db.BeginRo(ctx)
```

Comment:
- you already have `temptx` above
- `api.db` it's `chaindata` db, it's not related to txpool's db. So, "pre-state" of which db do you need? I don't see any "logging below"

Reply:
Oh, my bad. Thank you, updated [here](https://github.com/ledgerwatch/erigon/pull/8229/commits/35915f9ee0152af8b7d4df67864c534ed79c48ca).
Yes, I need the `chaindata` db.